Revisiting of Environmental variables precendence (#15396)
docker/docker.github.io: compose/envvars-precedence.md

---
title: Environment variables precedence
description: Scenario Overview illustrating how environmental variables are resolved in Compose
keywords: compose, environment, env file
---

## Order of precedence:

> 1. Passed from the command-line [`docker compose run --env <KEY[=VAL]>`](../../engine/reference/commandline/compose_run/#options).

Reviewer: ```suggestion
1. Passed from the command line [`docker compose run --env <KEY[=VAL]>`](../../engine/reference/commandline/compose_run/#options).
```

2. Passed from/set in `compose.yaml` service's configuration, [from the environment key](../../compose/compose-file/#environment).
3. Passed from/set in `compose.yaml` service's configuration, [from the env_file key](../../compose/compose-file/#env_file).
4. Passed from/set in Container Image [in ENV directive](../../engine/reference/builder/#env).

> Using as example `TAG`, a environmental variable defining the version for an image.

Reviewer: ```suggestion
`TAG`, an environmental variable defining the version for an image, is used as an example.
```

> Key to table, description for each column:

Reviewer: ```suggestion
Description for each column:
```

* Image - `ENV` directive in the Dockerfile
* .env file - `.env` file on the project root (or, with higher precedence, the file passed via `docker compose -–env-file <FILE>`).
* Command line - environmental variable passed via `docker compose run -e `<KEY[=VAL]>`.
* Compose file - in `environment` key from the service section in the `compose.yaml`.
* Compose file - in `env_file key` from the service section in the `compose.yaml`.

<!--
### Passed or Set
explanation here
 -->

> The table below aims to provide a quick overview of how interpolation works when using all environment variables on Compose.

Reviewer: ```suggestion
The table below provides a quick overview of how interpolation works when using all environment variables on Compose.
```
